lib/components/SelectionControl/radio/radio-button/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React from "react";
+import clsx from "clsx";
+import {
+    LabelPairedCircleDotMdFillIcon,
+    LabelPairedCircleDotSmFillIcon,
+    LabelPairedCircleInfoMdRegularIcon,
+    LabelPairedCircleInfoSmRegularIcon,
+    LabelPairedCircleMdRegularIcon,
+    LabelPairedCircleSmRegularIcon,
+} from "@deriv/quill-icons";
+import { Text } from "../../../Typography";
+import "./radio-button.scss";
+
+interface IRadio {
+    className?: string;
+    classNameInfo?: string;
+    classNameLabel?: string;
+    defaultChecked?: boolean;
+    disabled?: boolean;
+    has_info?: boolean;
+    id?: string;
+    name?: string;
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    required?: boolean;
+    value?: string;
+    size?: "sm" | "md";
+}
+
+const RadioButton = ({
+    children,
+    className,
+    classNameInfo,
+    classNameLabel,
+    defaultChecked,
+    disabled = false,
+    has_info,
+    id,
+    onChange,
+    size = "md",
+    value,
+    ...otherProps
+}: React.PropsWithChildren<IRadio>) => {
+    const [checked, setChecked] = React.useState(defaultChecked);
+
+    React.useEffect(() => {
+        setChecked(defaultChecked);
+    }, [defaultChecked]);
+
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setChecked(e.target.checked);
+        onChange?.(e);
+    };
+
+    const handleIconClick = (
+        e:
+            | React.MouseEvent<HTMLSpanElement>
+            | React.KeyboardEvent<HTMLSpanElement>,
+    ) => {
+        if (!disabled) {
+            const synthesizedEvent = new Event("change");
+            const input = e.currentTarget.previousSibling as HTMLInputElement;
+            input.dispatchEvent(synthesizedEvent);
+        }
+    };
+
+    const getIcon = () => {
+        if (checked) {
+            return size === "sm" ? (
+                <LabelPairedCircleDotSmFillIcon
```

Comment:
Seems that in Figma designers used StandAlone icons, not LabelPaired for radio btn.
`StandaloneCircleDotFillIconquill` icon component accept `size`, so there will be no need in passing width and height.
![Screenshot 2024-04-09 at 12 26 11 PM](https://github.com/deriv-com/quill-ui/assets/121025168/657a6473-452a-441a-aaaf-a78d0662a1ce)

Reply:
Please, ignore if irrelevant

Reply:
I'm not sure if this is the right way designer used because it's supposed to be an icon with a label, not standalone but since it has a size property, I switched to it. But again, designers shouldn't use standalone icons in a design with label. This defeats the purpose of having separate icons which are visually identical.